x-pack/plugin/esql/qa/testFixtures/src/main/resources/spatial-grid.csv-spec

```diff
@@ -0,0 +1,938 @@
+;
+
+geohexLiteralMv
+required_capability: spatial_grid
+
+ROW location = TO_GEOPOINT("POINT(12.6493508684508 55.6285017221528)"), precision = [1,2,3,4,5]
+| MV_EXPAND precision
+| EVAL geohex = ST_GEOHEX(location, precision)
+| EVAL geohexString = ST_GEOHEX_TO_STRING(geohex)
+;
+
+location:geo_point                        | precision:integer | geohex:long        | geohexString:keyword
+POINT (12.6493508684508 55.6285017221528) | 1                 | 581514107744681983 | 811f3ffffffffff
+POINT (12.6493508684508 55.6285017221528) | 2                 | 586013859081355263 | 821f07fffffffff
+POINT (12.6493508684508 55.6285017221528) | 3                 | 590517321269772287 | 831f05fffffffff
+POINT (12.6493508684508 55.6285017221528) | 4                 | 595020895127339007 | 841f059ffffffff
+POINT (12.6493508684508 55.6285017221528) | 5                 | 599524487238516735 | 851f0583fffffff
+;
+
+geohexField
+required_capability: spatial_grid
+
+FROM airports
+| WHERE abbrev == "CPH"
+| EVAL geohex = ST_GEOHEX(location, 7)
+| EVAL geohexString = ST_GEOHEX_TO_STRING(geohex)
+| KEEP geohex, geohexString, abbrev, name, location
+;
+
+geohex:long        | geohexString:keyword | abbrev:keyword | name:text  | location:geo_point
+608531685838946303 | 871f05818ffffff      | CPH            | Copenhagen | POINT (12.6493508684508 55.6285017221528)
+;
+
+gridGeohexStatsBy
+required_capability: spatial_grid
+
+// tag::st_geohex-grid[]
+FROM airports
+| EVAL geohex = ST_GEOHEX(location, 1)
+| STATS
+    count = COUNT(*),
+    centroid = ST_CENTROID_AGG(location)
+      BY geohex
+| WHERE count >= 10
+| EVAL geohexString = ST_GEOHEX_TO_STRING(geohex)
+| KEEP count, centroid, geohexString
+| SORT count DESC, geohexString ASC
+// end::st_geohex-grid[]
+;
+
+// tag::st_geohex-grid-result[]
+count:long | centroid:geo_point                             | geohexString:keyword 
+22         | POINT (7.250850197689777 48.21363834643059)    | 811fbffffffffff
+18         | POINT (-80.64959161449224 40.04119813675061)   | 812abffffffffff
+17         | POINT (-0.7606179875266903 52.86413913565304)  | 81197ffffffffff
+13         | POINT (22.53157936179867 41.98255742864254)    | 811efffffffffff
+13         | POINT (78.30096947387435 26.073904778951636)   | 813dbffffffffff
+12         | POINT (-76.39781514415517 45.16300531569868)   | 812bbffffffffff
+12         | POINT (-100.30120467301458 20.114154297625646) | 8149bffffffffff
+11         | POINT (18.037187419831753 48.66540593306788)   | 811e3ffffffffff
+11         | POINT (-83.42379064553164 33.18388901439241)   | 8144fffffffffff
+11         | POINT (-99.4237939513881 27.100012352774765)   | 8148bffffffffff
+10         | POINT (128.01009018346667 35.8699960866943)    | 8130fffffffffff
+// end::st_geohex-grid-result[]
+;
+
+gridGeohexQuery
+required_capability: spatial_grid
+
+FROM airports
+| WHERE ST_GEOHEX(location, 1) == ST_GEOHEX_TO_LONG("812bbffffffffff")
```

Comment:
Can we have a test for geo_shape as well? I expect it does not work for that as it is an equality on multivalues.

Reply:
I listed geo_shape as something to support in a followup PR. I did implemented the evaluators in a mv-friendly way, so it would be easier to support geo_shape, but right now this feature is geo_point only.